### PR TITLE
Change Nutanix CAPX to v1.3.0

### PR DESCRIPTION
**Pull request by adiantum**
*Description of changes:*
Nutanix CAPX changed to version v1.3.0 with default CCM component. Changed code, manifests and tests.
The change unblock ability to implement external etcd feature for Nutanix provider.

*Testing (if applicable):*

**Create cluster:**
```
$ eksctl anywhere create cluster -f ./eksa-test-add-ccm.yaml --bundles-override bin/local-bundle-release.yaml
Performing setup and validations
odd number of arguments passed as key-value pairs for logging	{"ignored key": "test-add-ccm"}
ValidateClusterSpec for Nutanix datacenter
Warning: Skipping TLS validation for insecure connection to Nutanix Prism Central; this is not recommended for production use
✅ Nutanix Provider setup is valid
✅ Validate OS is compatible with registry mirror configuration
✅ Validate certificate for registry mirror
✅ Validate authentication for git provider
✅ Validate cluster's eksaVersion matches EKS-A version
Creating new bootstrap cluster
Provider specific pre-capi-install-setup on bootstrap cluster
Installing cluster-api providers on bootstrap cluster
Provider specific post-setup
Creating new workload cluster
Installing networking on workload cluster
Creating EKS-A namespace
Installing cluster-api providers on workload cluster
Installing EKS-A secrets on workload cluster
Installing resources on management cluster
Moving cluster management from bootstrap to workload cluster
Installing EKS-A custom components (CRD and controller) on workload cluster
Installing EKS-D components on workload cluster
Creating EKS-A CRDs instances on workload cluster
Installing GitOps Toolkit on workload cluster
GitOps field not specified, bootstrap flux skipped
Writing cluster config file
Deleting bootstrap cluster
🎉 Cluster created!
--------------------------------------------------------------------------------------
The Amazon EKS Anywhere Curated Packages are only available to customers with the
Amazon EKS Anywhere Enterprise Subscription
--------------------------------------------------------------------------------------
Enabling curated packages on the cluster
Installing helm chart on cluster	{"chart": "eks-anywhere-packages", "version": "0.0.0-3b5440427449629eeb5c981bdd5743d37096c4c5"}

$ kubectl get clusterresourceset -n eksa-system
NAME                           AGE
test-add-ccm-nutanix-ccm-crs   9m32s

$ kubectl get nodes
NAME                            STATUS   ROLES           AGE   VERSION
test-add-ccm-25g7c              Ready    control-plane   11m   v1.28.4-eks-d91a302
test-add-ccm-4slvd              Ready    control-plane   12m   v1.28.4-eks-d91a302
test-add-ccm-hq4z5              Ready    control-plane   14m   v1.28.4-eks-d91a302
test-add-ccm-md-0-vld5n-p7zq8   Ready    <none>          13m   v1.28.4-eks-d91a302
test-add-ccm-md-0-vld5n-qvbxh   Ready    <none>          13m   v1.28.4-eks-d91a302
test-add-ccm-md-0-vld5n-vx7ls   Ready    <none>          13m   v1.28.4-eks-d91a302
```

**Upgrade cluster from v0.18.6:**
```
$ eksctl anywhere upgrade cluster -f ./test-update/eksa-test-update-ccm.yaml --bundles-override bin/local-bundle-release.yaml
Performing setup and validations
✅ Nutanix provider validation
✅ SSH Keys present
✅ Validate OS is compatible with registry mirror configuration
✅ Validate certificate for registry mirror
✅ Control plane ready
✅ Worker nodes ready
✅ Nodes ready
✅ Cluster CRDs ready
✅ Cluster object present on workload cluster
✅ Upgrade cluster kubernetes version increment
✅ Upgrade cluster worker node group kubernetes version increment
✅ Validate authentication for git provider
✅ Validate immutable fields
✅ Validate cluster's eksaVersion matches EKS-Anywhere Version
✅ Validate pod disruption budgets
✅ Validate eksaVersion skew is one minor version
Ensuring etcd CAPI providers exist on management cluster before upgrade
Pausing GitOps cluster resources reconcile
Upgrading core components
Backing up management cluster's resources before upgrading
Upgrading management cluster
Updating Git Repo with new EKS-A cluster spec
GitOps field not specified, update git repo skipped
Forcing reconcile Git repo with latest commit
GitOps not configured, force reconcile flux git repo skipped
Resuming GitOps cluster resources kustomization
Writing cluster config file
🎉 Cluster upgraded!
Cleaning up backup resources

$ kubectl get clusterresourceset -n eksa-system
NAME                              AGE
test-update-ccm-nutanix-ccm-crs   18m

$ kubectl get nodes
NAME                               STATUS   ROLES           AGE     VERSION
test-update-ccm-2wtjs              Ready    control-plane   12m     v1.28.4-eks-d91a302
test-update-ccm-dhqcp              Ready    control-plane   16m     v1.28.4-eks-d91a302
test-update-ccm-jtllt              Ready    control-plane   17m     v1.28.4-eks-d91a302
test-update-ccm-md-0-9dbsk-ktpbh   Ready    <none>          7m25s   v1.28.4-eks-d91a302
test-update-ccm-md-0-9dbsk-lb7hj   Ready    <none>          6m32s   v1.28.4-eks-d91a302
test-update-ccm-md-0-9dbsk-mggsb   Ready    <none>          8m40s   v1.28.4-eks-d91a302
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by abhinavmpandey08 — COMMENTED** [review]
awesome work! PR looks mostly good, let a few small comments.
Also, can you add the CCM image to `NutanixImages` here? https://github.com/abhinavmpandey08/eks-anywhere/blob/664bac4a90e20e440105fb03862e83e108841bd1/release/api/v1alpha1/artifacts.go#L154-L160

File: pkg/providers/nutanix/config/cp-template.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metadata:
   name: "{{.clusterName}}"
   namespace: "{{.eksaSystemNamespace}}"
 spec:
+  failureDomains: []
```
[the remark this answers is not in the document]
Comment (adiantum) [reply pkg/providers/nutanix/config/cp-template.yaml:7]: @abhinavmpandey08 Unfortunately it's required. In CAPX v1.3.0 introduced failure domains, but there is missed `omitempty` directive for the field. Without this definition the error raised.

**Review by abhinavmpandey08 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**